chainer/functions/activation/sigmoid.py

```diff
@@ -12,51 +12,88 @@
     _mode = libcudnn.CUDNN_ACTIVATION_SIGMOID
 
 
-class Sigmoid(function.Function):
+class Sigmoid(function_node.FunctionNode):
 
     """Logistic sigmoid function."""
 
     def check_type_forward(self, in_types):
         type_check.expect(in_types.size() == 1)
         type_check.expect(in_types[0].dtype.kind == 'f')
 
-    def forward_cpu(self, x):
-        half = x[0].dtype.type(0.5)
-        y = utils.force_array(numpy.tanh(x[0] * half) * half + half)
+    def forward_cpu(self, inputs):
+        x = inputs[0]
+        half = x.dtype.type(0.5)
+        y = utils.force_array(numpy.tanh(x * half) * half + half)
+        self.retain_inputs(())
```

Comment:
This `retain_inputs` is not needed anymore (inputs are not retained by default in FunctionNode).

Reply:
delete it